agent/virustotal.py

```diff
@@ -55,7 +58,10 @@ def get_scans(response: dict[str, Any]) -> dict[str, Any] | None:
     Raises:
         VirusTotalApiError: In case the API request encountered problems.
     """
-    if response["response_code"] == 0 or "results" not in response:
+    if response.get("response_code") == 204 and response.get("error") is not None:
+        logger.error("Exceeded the virustotal API rate limit.")
+        raise VirusTotalApiError()
```

Comment:
And we crash???

Reply:
We do catch it outside.

Reply:
You should remove the raise exception in _process_response when you catch VirusTotalApiError